hypervisor/include/arch/x86/guest/vlapic.h

```diff
@@ -74,29 +69,19 @@ struct acrn_vlapic {
 	struct vlapic_pir_desc	pir_desc;
 
 	struct acrn_vm		*vm;
-	struct acrn_vcpu		*vcpu;
+	struct acrn_vcpu	*vcpu;
 
 	uint32_t		esr_pending;
 	int32_t			esr_firing;
 
 	struct vlapic_timer	vtimer;
 
 	/*
-	 * The 'isrvec_stk' is a stack of vectors injected by the local apic.
-	 * A vector is popped from the stack when the processor does an EOI.
-	 * The vector on the top of the stack is used to compute the
-	 * Processor Priority in conjunction with the TPR.
-	 *
-	 * Note: isrvec_stk_top is unsigned and always equal to the number of
-	 * vectors in the stack.
-	 *
-	 * Operations:
-	 *     init: isrvec_stk_top = 0;
-	 *     push: isrvec_stk_top++; isrvec_stk[isrvec_stk_top] = x;
-	 *     pop : isrvec_stk_top--;
+	 * isr_nr: vector numbers that are set in the ISR
+	 * isrv: vector number for the highest priority bit that is set in the ISR
 	 */
-	uint8_t		isrvec_stk[ISRVEC_STK_SIZE];
-	uint32_t	isrvec_stk_top;
+	uint16_t	isr_nr;
+	uint32_t	isrv;
```

Comment:
not very clear why we need isr_nr and isrv here?

Reply:
For APICv Base case, there're too many place needs to update PPR in order to support TPR below threshold. Updating PPR needs ISRV. But there're only two place needs to update ISRV (one is deliver IRRV to ISR, another is handle EOI). As for isr_nr, it's waste of time to walk through ISRs array to calculate ISRV if there's no bit is set in ISRs. However, cache isr_nr is not expensive. Both of them are for speeding up the update the TPR.

Reply:
even the current code of isr_nr is incorrect. If the guest write EOI mulitple times, isr_nr may becomes 0xFFFFFFFFF.

I am not convinced.

Reply:
It couldn't since if the guest write EOI multiple times the isrv will become zero.  In this way, we could make sure the isr_nr will not become 0xFFFFFFFF.
I will remove it if you don't think we need it.